storage/innobase/srv/srv0srv.cc

```diff
@@ -2675,6 +2681,73 @@ srv_master_do_shutdown_tasks(
 	return(n_bytes_merged || n_tables_to_drop);
 }
 
+void
+srv_enable_temp_encryption_if_set()
+{
+	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
+		return;
+	}
+
+	if (is_shared_system_tablespace(srv_tmp_space.space_id())) {
+		/* there is no separate tablespace for temporary tables */
+		return;
+	}
+
+	if (srv_tmp_tablespace_encrypt) {
+
+		if (srv_read_only_mode) {
+			srv_tmp_tablespace_encrypt = false;
+			ib::error() << "Can't set temporary tablespace to be"
+				<< " encrypted in read-only mode.";
+			return;
+		}
+
+		/* Make sure the keyring is loaded. */
+		if (!Encryption::check_keyring()) {
+			srv_tmp_tablespace_encrypt = false;
+			ib::error() << "Can't set temporary tablespace to be"
+				<< " encrypted because keyring plugin is"
+				<< " not available.";
+			return;
+		}
+
+		fil_space_t* space = fil_space_get(srv_tmp_space.space_id());
+
+		ut_ad(fsp_is_system_temporary(space->id));
+
+		if (!FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
+
+			if (!fsp_enable_encryption(srv_tmp_space.space_id())) {
+				srv_tmp_tablespace_encrypt = false;
+				ib::warn() << "Can't set temporary"
+					<< " tablespace to be"
+					<< " encrypted.";
+				return;
+			}
+
+#ifdef UNIV_DEBUG
+			if (srv_master_encrypt_debug != 0) {
+				srv_master_encrypt_debug = 2;
+				while (srv_master_encrypt_debug != 0) {
+					os_thread_sleep(10000);
+				}
+			}
+#endif
+
+			space->flags |= FSP_FLAGS_MASK_ENCRYPTION;
+
+			if (fil_set_encryption(space->id,
```

Comment:
Looking at what fil_set_encryption may return under what conditions, it looks like this check can be replaced by an assert?

Reply:
replaced by assert